src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -16,9 +16,9 @@ namespace Microsoft.CodeAnalysis.CSharp
     /// </summary>
     internal sealed class InMethodBinder : LocalScopeBinder
     {
-        private readonly MultiDictionary<string, ParameterSymbol> _parameterMap;
+        private MultiDictionary<string, ParameterSymbol> _lazyParameterMap;
```

Comment:
This isn't specific to switch, it affects other statements too. Creation of a binder should be a very cheap operation that doesn't trigger any binding. Otherwise, it is possible to get into an infinite cycle. Creation of a binder -> binding -> creation of a local -> lookup for a binder -> creation for the binder again. This binder is used to handle local functions and its constructor was doing binding.